dali/server/daldap.cpp

```diff
@@ -225,9 +226,10 @@ class CDaliLdapConnection: implements IDaliLdapConnection, public CInterface
 
         if (!authenticated)
         {
-            user->credentials().setPassword(password);
-            if (!ldapsecurity->authenticateUser(*user, &superUser) || !superUser)
+            CLdapSecManager* secmgr = dynamic_cast<CLdapSecManager*>(ldapsecurity.get());
+            if (!secmgr->isSuperUser(user))
```

Comment:
@RussWhitehead the dynamic cast may have failed, in which case, for now at least I think access should be denied.

So perhaps "if (!secmgr || !secmgr->isSuperUser(user))"

Also, personally I would call the variable ldapmgr since you've casted to an ldap specific interface/class.

Reply:
@afishbeck Done